### ISSUE TITLE
How does reflow sentence_per_line interact with line length?

### ISSUE THREAD

**Issue by ahgraber** [rvben/rumdl#111]
I'm experimenting with reflow settings (in case you could not tell from other issues - side note: reflow is a fantastic concept and I'm super pleased with it regardless of my other bug reports).
How does sentence_per_line interact with the global and MD013 line length settings? I can't seem to get it to trigger a sentence-per-line reflow reliably, and if I do it myself then I get error reports on line length. My interpretation of "sentence_per_line" is that it automates breaking on sentence boundaries _regardless of configured line lengths_ - i.e., it would ignore line length warnings if the line contains a single sentence. But maybe I'm misinterpreting!

```toml
# Set global line length (used by MD013 and other line-length rules)
line_length = 120

[MD013]
line_length = 120
code_blocks = false # Don't check code blocks (default: true)
# code_block_line_length = 119
headings = true
# headings_line_length = 80
tables = false
# stern = false # Stricter checking without exceptions
# strict = false # Disables exceptions for URLs, etc.
reflow = true # Enable automatic text reflow/wrapping (default: false)
reflow_mode = "sentence_per_line" # Reflow mode: "default", "normalize", or "sentence_per_line" (default: "default")
```

**Comment by rvben** [comment]
Hi @ahgraber,

Excellent question! You've identified an important design issue with `sentence_per_line` mode. After reviewing your feedback, I agree that the original behavior was counterintuitive.

## What Changed

I've just updated `sentence_per_line` mode to work more intuitively. Now when you set `line_length = 120`, it actually **enforces** that limit, even in `sentence_per_line` mode.

## New Behavior

With your configuration:
```toml
[MD013]
line_length = 120
reflow = true
reflow_mode = "sentence_per_line"
```

| Scenario | Result | Auto-Fixable? |
|----------|--------|---------------|
| Single sentence, 50 chars | ✅ No violation | N/A |
| Single sentence, 200 chars | ⚠️ **Warning** | ❌ No |
| Two sentences, 80 chars total | ⚠️ **Warning** | ✅ Yes |

## Example

```markdown
# This now triggers a warning (but cannot be auto-fixed)
This is a very long sentence that goes on and on for 234 characters, way beyond your 120-character limit, and will now trigger an MD013 warning because it exceeds the configured line length.

# This triggers a warning AND can be auto-fixed
This is sentence one. This is sentence two on the same line.
```

After running `rumdl check`:
```
file.md:3:121: [MD013] Line length 234 exceeds 120 characters
file.md:5:1: [MD013] Line contains multiple sentences [*]

Issues: Found 2 issues in 1 file
Run `rumdl fmt` to automatically fix 1 of the 2 issues
```

After running `rumdl check --fix`:
```markdown
# Long single sentence - unchanged (you must manually rephrase)
This is a very long sentence that goes on and on for 234 characters, way beyond your 120-character limit, and will now trigger an MD013 warning because it exceeds the configured line length.

# Multiple sentences - automatically split
This is sentence one.
This is sentence two on the same line.
```

The warning for the long single sentence **remains** after auto-fix because splitting mid-sentence would break the meaning. You need to manually rephrase it.

## Why This Design?

This new approach is more intuitive because:

1. **Respects your configuration**: If you set `line_length = 120`, you probably care about that limit
2. **Semantic integrity**: Won't auto-fix by splitting mid-sentence (that would be wrong)
3. **Clear feedback**: Tells you which lines need manual attention (no `[*]`) vs. automatic fixing (`[*]`)

## The Three Reflow Modes

For reference, here's how the modes differ:

| Mode | Line Length Enforcement | Sentence Boundaries |
|------|------------------------|---------------------|
| `default` | ✅ Yes (wraps at length) | ❌ Ignores |
| `normalize` | ✅ Yes (combines short lines, wraps long ones) | ❌ Ignores |
| `sentence_per_line` | ✅ Yes (warns on long single sentences) | ✅ Splits multi-sentence lines |

## Bottom Line

Your interpretation was spot on - `line_length` **should** matter in `sentence_per_line` mode. The original implementation was indeed counterintuitive.

The fix is now in the codebase and will be included in the next release. Thanks for the great question that led to this improvement!

Let me know if you have any other questions about reflow modes.
